### DISCUSSION TITLE
How can I create an async Aspect?

### DISCUSSION THREAD

**Discussion by fcastells** [pamidur/aspect-injector#229] [Q&A]
I need to run async code around (before and after) a method. I can run code both by creating an Aspect and by creating a derived attribute from one of the Universal attributes. Both options allow me to decorate an async function (a function returning a Task), but none of these options allow me (at least not that I found it) to run async code in the aspect itself. 

`BaseUniversalWrapperAttribute` has a virtual  WrapAsync<T> function that would allow me to do what I need, but for some reason it is marked as internal so I can't override it. 

Is there a way currently to run async code in an aspect?

Thanks

**Comment by pamidur** [comment]
Hey @fcastells , sorry for long delay , please have a look here if still needed https://github.com/pamidur/aspect-injector/discussions/243
